### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by dicodiaz**
**Implemented features:**
- Create a `current_user` method in `ApplicationController` to make the current user data available to all controllers
    - It returns the first user from the database
- Create forms to perform the following functions:
    - Create a Post on behalf of the `current_user`
    - Create a comment on behalf of the `current_user`
- Allow Users to add likes to Posts

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    comment = Comment.new(params.require(:comment).permit(:post_id, :text))
+    comment.author = current_user
+    return unless comment.save
+
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:id])
+    @liked = Like.exists?(author: current_user, post: @post)
+    render template: 'posts/show'
+  end
+end
```
Comment (CollinsTatang) [inline app/controllers/comments_controller.rb]: - [x] Please your comment controller should have a `new` and `create` action method. Kindly add the new action method.
```ruby
 ...

  def new
    @comment = Comment.new
  end
```

- [x] Please there should be a form in the views/comments/new.html.erb file which accepts input for creating comments.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -6,5 +6,21 @@ def index
...
     @post = Post.find(params[:id])
+    @liked = Like.exists?(author: current_user, post: @post)
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(params.require(:post).permit(:title, :text))
```
Comment (CollinsTatang) [inline app/controllers/posts_controller.rb]: ### **Optional**

- [x] Please you can further secure your input by using the [Strong  Parameter](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md)

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @dicodiaz, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (dicodiaz) [reply app/controllers/posts_controller.rb]: Hi @CollinsTatang, thanks for your review! Please correct me if I'm wrong, but I think I'm using strong parameters by adding `params.require(:post).permit(:title, :text)`. Do you mean moving this line to a private method?

Comment (CollinsTatang) [reply app/controllers/posts_controller.rb]: Yes @dicodiaz 
You can check out this [guide](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md) for more info.

Comment (dicodiaz) [reply app/controllers/posts_controller.rb]: Gotcha, thank you very much

**Review by DuaneDave — APPROVED** [review]
Hi @dicodiaz 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good job adding the forms
✔️ Good job with the styling of the application

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**